### PR TITLE
[MPS] Add native ops fast path for SDPA on long sequences (~2x speedup)

### PR DESCRIPTION
**Pull request by baisampayans**
## Summary

`F.scaled_dot_product_attention` on MPS is **~2x slower** than equivalent manual `at::matmul` + `at::softmax` for sequences longer than 8 tokens. This PR adds a fast path that bypasses the MPSGraph-based `sdpa_general_mps` and calls native MPS ops directly, eliminating graph compilation/dispatch overhead.

### Root Cause

For `query_seq_len > 8`, `_scaled_dot_product_attention_math_mps` falls through to `sdpa_general_mps`, which:
1. Builds an MPSGraph computation graph (`matmul -> upcast_to_fp32 -> scale -> softmax -> matmul`)
2. Upcasts attention scores to float32 (doubling bandwidth)
3. Allocates and returns a full `[B, H, S, S]` attention weight matrix (even when unused)
4. Applies zero-mask correction for all-negative-infinity rows (5 extra graph ops)

The native `at::matmul` path goes through MPS's optimized BLAS kernels with no graph overhead.

### Changes

- Added 2 includes (`ATen/ops/matmul.h`, `ATen/ops/softmax.h`)
- Added 13-line fast path in `_scaled_dot_product_attention_math_mps` for the common case: `!is_causal && !mask && query_seq_len > 8`
- Existing short-sequence paths (`sdpa_vector_fast_mps`, `sdpa_vector_2pass_mps`) are unaffected

### Benchmarks

**Isolated attention benchmark** (M3 Ultra, 76 GPU cores, bf16, PyTorch built from this branch):

| Shape | Before | After | Speedup |
|---|---|---|---|
| `[1, 24, 4608, 128]` (Flux2 DiT) | 37.8ms | 19.0ms | **1.99x** |
| `[1, 24, 1024, 128]` | 2.2ms | 1.5ms | **1.47x** |
| `[1, 32, 4096, 128]` (Llama-style) | 39.2ms | 19.8ms | **1.98x** |
| `[1, 12, 2048, 64]` (ViT-Large) | 3.5ms | 1.7ms | **2.06x** |
| `[1, 8, 8192, 128]` (Long-context) | 40.6ms | 20.5ms | **1.98x** |
| `[1, 40, 2048, 128]` (Llama-70B) | 12.4ms | 6.4ms | **1.94x** |
| `[2, 24, 4608, 128]` (batch=2) | 81.6ms | 36.6ms | **2.23x** |

**End-to-end on real models:**
- FLUX.2 Klein 9B image generation (1024x1024, 2-step): 5.88s → **4.62s** (1.27x)
- LTX-Video 2B video generation (512p, 4s, 4-step): 13.4s → **11.4s** (1.18x)

**Numerical accuracy:** Max absolute difference vs. `sdpa_general_mps` = **0.000000** (exact match on bf16).

### Standalone Repro Script

```python
import torch, time
import torch.nn.functional as F

device, dtype = "mps", torch.bfloat16
B, H, S, D = 1, 24, 4608, 128

q = torch.randn(B, H, S, D, device=device, dtype=dtype)
k = torch.randn(B, H, S, D, device=device, dtype=dtype)
v = torch.randn(B, H, S, D, device=device, dtype=dtype)

def bench(fn, name, warmup=3, repeats=10):
    for _ in range(warmup): fn()
    torch.mps.synchronize()
    times = []
    for _ in range(repeats):
        torch.mps.synchronize()
        t0 = time.perf_counter()
        fn()
        torch.mps.synchronize()
        times.append((time.perf_counter() - t0) * 1000)
    print(f"{name}: {sum(times)/len(times):.2f}ms")

bench(lambda: F.scaled_dot_product_attention(q, k, v), "SDPA")
```

**Before this PR:** ~37ms. **After:** ~19ms.

### Related Issues
- #163597 — SDPA MPS regression on 2.8.0 (non-contiguous tensor bug, different issue but related dispatch path)
- #135778 — Severe SDPA Performance Regression 2.5.0-RC1
- #155797 — Performance regression with ComfyUI Flux on MPS
- Forum: [SDPA slower than matmul](https://discuss.pytorch.org/t/bug-scaled-dot-product-attention-slower-than-matmul/223525)

cc @kulinseth @malfet @DenisVieriu97

### CODE REVIEW

**Comment by hvaara** [comment]
Just looked into it a bit more. This PR currently fails on `test/test_transformers.py::TestSDPAMPS::test_scaled_dot_product_attention_fp16_overflow_mps`, which is a regression test for #160841 (fp16 overflow). See my previous comment https://github.com/pytorch/pytorch/pull/178545#issuecomment-4176006157.

@baisampayans I don't think you'll be able to pass the tests without qkv upcast similar to the native implementation, and in that case most of the speedup is likely gone. Calling `at::native::_scaled_dot_product_attention_math` directly results in about a 10% performance degradation compared to MPSGraph. Potentially it's recoverable elsewhere (inefficiencies downstream via the ops you call). Personally I would favor a 10% perf degradation to migrate off MPSGraph, but I'm not sure that's the predominant opinion.

**Comment by hvaara** [comment]
@baisampayans Just to keep you in the loop, there are several ongoing discussion about SDPA on MPS at the moment. #179294 is a good entrypoint for an overall understanding, and also links to other issues with the current implementation in case you're interested in reading up on it. I'd just hold off for now, and see what they decide 😃
